src/PlaywrightLauncher.js

```diff
@@ -269,4 +271,46 @@ class PlaywrightLauncher {
   }
 }
 
+async function loadScopesFromManifestStrict({ connectorPath, __dirname, log }) {
+  const manifestPath = path.resolve(__dirname, '..', connectorPath, 'manifest.konnector');
+  log(`🔎 Reading manifest: ${manifestPath}`);
+  let raw;
+  try {
+    raw = await fs.promises.readFile(manifestPath, 'utf8');
+  } catch (err) {
+    log(`❌ Cannot read manifest file: ${err.message}`);
+    throw new Error(`Manifest not found or unreadable at ${manifestPath}`);
+  }
+
+  let manifest;
+  try {
+    manifest = JSON.parse(raw);
+  } catch (err) {
+    log(`❌ Invalid JSON manifest: ${err.message}`);
+    throw new Error(`Manifest JSON invalid at ${manifestPath}`);
+  }
+
+  const perms = manifest.permissions;
```

Comment:
We could imagine a connector without permissions. I prefer not to cause an error in this case. Let's set perms to [] by default.